cfme/tests/containers/test_manageiq_ansible_custom_attributes.py

```diff
@@ -0,0 +1,141 @@
+import pytest
+from cfme.fixtures import pytest_selenium as sel
+from cfme.containers.provider import ContainersProvider
+from utils import testgen
+from utils.ansible import setup_ansible_script, run_ansible, \
+    fetch_miq_ansible_module, revert_basic_script
+from utils.version import current_version
+
+
+pytestmark = [
+    pytest.mark.uncollectif(lambda provider: current_version() < "5.7")]
+pytest_generate_tests = testgen.generate([ContainersProvider], scope='function')
+
+custom_attributes_to_add = {
+    'name': 'custom1',
+    'value': 'first value'
+}, {
+    'name': 'custom2',
+    'value': 'second value'
+}
+
+
+custom_attributes_to_edit = {
+    'name': 'custom1',
+    'value': 'third value'
+}, {
+    'name': 'custom2',
+    'value': 'fourth value'
+}
+
+provider_name = 'CI OSE'
+
+
+@pytest.yield_fixture(scope='function')
+def ansible_custom_attributes():
+    fetch_miq_ansible_module()
+    yield
+    revert_basic_script(script_type='custom_attributes')
+
+
+@pytest.mark.polarion('CMP-10559')
+@pytest.mark.usefixtures('setup_provider')
+def test_manageiq_ansible_add_custom_attributes(ansible_custom_attributes, provider):
+    """This test checks adding a Custom Attribute using Ansible script via Manage IQ module
+        Steps:
+        1. 'add_custom_attributes.yml script runs against the appliance
+         and adds custom attributes
+        2. Test navigates  to Providers page and verifies the Custom Attributes
+         were added under Providers menu
+
+        """
+    setup_ansible_script(provider, script='add_custom_attributes',
+                         values_to_update=custom_attributes_to_add,
+                         script_type='custom_attributes')
+    run_ansible('add_custom_attributes')
+    for custom_attribute in custom_attributes_to_add:
+        assert provider.get_detail('Custom Attributes',
+                                   custom_attribute['name']) == custom_attribute['value']
+        assert sel.is_displayed_text('Custom Attributes')
+
+
+@pytest.mark.polarion('CMP-10560')
+def test_manageiq_ansible_edit_custom_attributes(ansible_custom_attributes, provider):
+    """This test checks editing a Custom Attribute using Ansible script via Manage IQ module
+        Steps:
+        1. 'add_custom_attributes.yml script runs against the appliance
+         and edits custom attributes
+        2. Test navigates to Providers page and verifies the Custom Attributes
+         were edited under Providers menu
+
+        """
+    setup_ansible_script(provider, script='add_custom_attributes',
+                         values_to_update=custom_attributes_to_edit,
+                         script_type='custom_attributes')
+    run_ansible('add_custom_attributes')
+    pytest.sel.refresh()
```

Comment:
**REQUIRED:** What is this refresh for? The navigation should be enough. We want to avoid calls to sel, so having it maybe even navigate to the provider again at the start of the test ,and then navigate to the details page is probably a good idea.